File: backend/app/routers/partnership.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.database import get_db
from app.schemas.partnership import PartnershipRequestCreate, PartnershipRequestOut
from app.services.partnership_service import PartnershipService

router = APIRouter(prefix="/api/partnership", tags=["partnership"])


@router.post("/request", response_model=PartnershipRequestOut, status_code=201)
def submit_request(data: PartnershipRequestCreate, db: Session = Depends(get_db)):
    return PartnershipService(db).create_request(data)

File: frontend/src/api/partnership.ts
import client from './client'
import type { PartnershipRequest } from '../types'

export const submitPartnershipRequest = (data: {
  company_name: string
  email: string
  phone: string
}) =>
  client.post<PartnershipRequest>('/api/partnership/request', data).then((r) => r.data)
